### ISSUE TITLE
PUT requests not handled correctly by the REST library

### ISSUE THREAD

**Issue by gotwarlost** [YahooArchive/mojito#429]
from rest.common.js:

``` javascript
                if ('GET' === method) {
                    if (-1 === url.indexOf('?')) {
                        url += '?' + params;
                    } else {
                        url += '&' + params;
                    }
                } else if ('POST' === method) { /* WHAT ABOUT PUT? */
                    ioConfig.data = params;
                }
```

This bug is not present in 0.3.26 and was introduced in 0.3.27. It is affecting our capability to move forward since we need PUT requests to a backend webservice to work.

**Comment by rwaldura** [comment]
Ouch.
https://github.com/yahoo/mojito/blob/develop/lib/app/autoload/rest.common.js#L75

**Comment by rwaldura** [comment]
Next Sprint candidate. Sprint's already pretty full though -- might get bumped. How critical is this? What's it blocking?

**Comment by mojit0** [comment]
A proper REST implementation should support GET, POST, PUT, and DELETE. For performance reasons it's sometimes useful to also support HEAD requests. As part of that support HTTP status codes other than 200 can indicate success and should be properly handled. We may want to open a separate issue for full REST support so we can focus this bug on only what's being blocked at the moment tho.

**Comment by gotwarlost** [comment]
agree with @mojit0 - currently we are unable to move forward with our mojito dependency and have to pin ourselves to v0.3.26

This particular fix should be a one-line change (i.e. add a clause for PUT in addition to POST) - the effort is more on writing a unit test for this.

**Comment by rwaldura** [comment]
Next Sprint, S41.

**Comment by mridgway** [comment]
Looks like this regression was added by #196 which was a fix for a YUI 3.5.1 regression. If we're upgrading to YUI 3.6 we can roll back #196 because it was fixed with IO.

**Comment by mridgway** [comment]
Upgrade to YUI 3.6 is blocked by a regression, so we're waiting until 3.6.1 or 3.7.0 before we upgrade. Therefore, I am fixing this in place as it is now. PR incoming.

**Comment by mridgway** [comment]
Fixed in #450.
